### ISSUE TITLE
Bug in IsomorphismPermGroup

### ISSUE THREAD

**Issue by james-d-mitchell** [gap-packages/Semigroups#637]
I get the following:
~~~~
gap> H := HClass(FullTransformationMonoid(4), Transformation([2, 2, 2]));
<Green's H-class: Transformation( [ 2, 2, 2 ] )>
gap> map := IsomorphismPermGroup(H);
MappingByFunction( <Green's H-class: Transformation( [ 2, 2, 2 ] )>, Group(
[ (2,4) ]), function( x ) ... end, function( x ) ... end )
gap> p :=Transformation( [ 2, 2, 2, 2 ] ) ^ map;

gap> OnTuples([1 .. 10], p);
[ 1, 2, 3, 2, 5, 6, 7, 8, 9, 10 ]
~~~~

The issue is that `Transformation( [ 2, 2, 2, 2] )` does not belong to the domain of `map` but this is not checked.

**Comment by james-d-mitchell** [comment]
Thanks to @zoltancoccyx for pointing this out.

**Comment by james-d-mitchell** [comment]
Another related bug is the following:

~~~~
H := HClass(FullTransformationMonoid(4), Transformation([3, 1, 2, 3]));
map := IsomorphismPermGroup(H);
BruteForceIsoCheck(map);
Transformation( [ 3, 1, 2, 3 ] ) Transformation( [ 3, 1, 2, 3 ] )
false
~~~~

**Comment by james-d-mitchell** [comment]
Resolved in v3.2.2
